src/content/getting-started/secure-your-app/_index.md

```diff
@@ -1,29 +1,159 @@
+
+## Step 3:  Create a policy exception
+
+Let’s assume the `hello-world` application needs to run as a `root` user because the container image is designed this way. You don't have more time and your CTO is pushing you to deploy the application.
+
+First, you need to modify the deployment to configure the `securityContext` to allow the container to run as a `root` user.
+
+```yaml
+spec:
+  containers:
+  - name: hello-world
+    ...
+    securityContext:
+      allowPrivilegeEscalation: false
+      capabilities:
+        drop:
+        - ALL
+      runAsNonRoot: false
+      runAsUser: 0
+      seccompProfile:
+```
+
+When you try to deploy the application, the policy engine will block the deployment with a message like:
+
+```text
+require-run-as-non-root-user:
+  autogen-run-as-non-root-user: 'validation error: Running as root is not allowed.
+    The fields spec.securityContext.runAsUser, spec.containers[*].securityContext.runAsUser,
+    spec.initContainers[*].securityContext.runAsUser, and spec.ephemeralContainers[*].securityContext.runAsUser
+    must be unset or set to a number greater than zero. rule autogen-run-as-non-root-user
+    failed at path /spec/template/spec/containers/0/securityContext/runAsUser/'
+```
+
+It works as expected. Now, it's time to create a policy exception. Depending on your platform team the policy has to be approved before it's applied to the cluster. In this guide, we will assume the policy is approved.
+
+In the platform, there is an extended API, `PolicyException`, to abstract the policy definition from the policy engine. If the underlying policy engine is changed, the `PolicyException` resource will be still valid. In our scenario, the resource looks like:
+
+```yaml
+apiVersion: policy.giantswarm.io/v1alpha1
+kind: PolicyException
+metadata:
+  name: hello-world-allow-root
+  namespace: policy-exceptions
+spec:
+  policies:
+  - require-run-as-non-root-user
+  - require-run-as-nonroot
+  targets:
+  - kind: Deployment
+    names:
+    - hello-world*
+    namespaces:
+    - default
+```
 
-## Step 1: To be determined
+This policy is translated to the policy engine as:
 
-<content>
+```yaml
+apiVersion: kyverno.io/v2beta1
+kind: PolicyException
+metadata:
+  name: hello-world-allow-root
+  namespace: policy-exceptions
+spec:
+  exceptions:
+  - policyName: require-run-as-non-root-user
+    ruleNames:
+    - run-as-non-root-user
+    - autogen-run-as-non-root-user
+    - autogen-cronjob-run-as-non-root-user
+  ...
+  match:
+    any:
+    - resources:
+        kinds:
+        - Deployment
+        - Pod
+        names:
+        - hello-world*
+        namespaces:
+        - default
+```
 
-## Step 2:  To be determined
+Note that the exception is created in the `policy-exceptions` namespace. This decision is made consciously to separate exceptions from workloads resources. This allows the security team to approve exceptions independently.
```

Comment:
@fhielpos WDYT now? I dont want to explain too much about the policy exception flow you would like to pursue